### PR TITLE
Feat/add member activity rights page

### PR DESCRIPTION
**Pull request by TedxTed**
# Activity Ticket Page Instructions

## Overview
Switch the usage from `order_product` route to the new `activity_ticket` route. Introduce a new page called `ActivityTicketDetailsPage` and employ the `activity/member-right` API. Additionally, a new hook named `activityTicket.ts` will be created.

### Change the Product Route
Modify the endpoint configuration in the product route as follows:

**File**: `src/helpers`  
**Line**: #506

**Original**:  
```javascript
`${process.env.REACT_APP_LODESTAR_SERVER_ENDPOINT}${route}${memberId ? `/?memberId=${memberId}` : ''}`,
```

**Updated**:  
```javascript
`https://demomo.kolable.dev/api/v2${route}${memberId ? `/?memberId=${memberId}` : ''}`,
```

### Set Up the Local Lodestar Server
- **Branch to use**: `feat/member-activity-ticket-right-api`

### Update the Lodestar Server Environment Route
Change the lodestar-server environment route to point to localhost on port 8081.

### Access the Member Page
Navigate to the member page at the following URL to view all the member's activity tickets:

**URL**:  
`http://localhost:3333/members/793733e7-c270-4db1-ae71-562984265429?tabkey=activity-ticket`

On this page, each link should direct to the `ActivityTicketDetailsPage`. There are two scenarios to test:

1. **No Session Query String**:  
   URL: `http://localhost:3333/activity_ticket/2965b630-1e86-4d05-a5f6-b2410768f5f1`

2. **With Session Query String**:  
   URL: `http://localhost:3333/activity_ticket/2965b630-1e86-4d05-a5f6-b2410768f5f1?session=39efb645-aea6-4a7b-9002-66449197aef1`

### CODE REVIEW

**Comment by TedxTed** [comment]
[(3)【活動】權益判斷另開API [2]](https://trello.com/c/Lvk4JOSt/5195-3%E3%80%90%E6%B4%BB%E5%8B%95%E3%80%91%E6%AC%8A%E7%9B%8A%E5%88%A4%E6%96%B7%E5%8F%A6%E9%96%8Bapi-2)

**Review by LoisChen68 — CHANGES_REQUESTED** [review]

**Review by LoisChen68 — CHANGES_REQUESTED** [review]

File: src/pages/MemberPage/ActivityTicketCollectionBlock.tsx [hunk 1]
```diff
@@ -49,6 +60,25 @@ const ActivityTicketCollectionBlock: React.VFC<{
     )
   }
 
+  const seenSessionIds = new Set()
+  const sessions = tickets
```
Comment (Ponchimeow) [inline src/pages/MemberPage/ActivityTicketCollectionBlock.tsx:64]: add `.filter(ticket => ticket.sessions.length > 0)`

File: src/hooks/activityTicket.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+          }
+        } catch (err) {
+          console.log(err)
+          setData(null)
+        } finally {
+          setLoading(false)
+        }
+      }
+    },
+    [currentMemberId, activityTicketId],
```
[the remark this answers is not in the document]
Comment (kkshyu) [reply src/hooks/activityTicket.ts]: FYI: https://github.com/xiaohesong/TIL/blob/master/front-end/react/overreact/%E4%B8%80%E4%BB%BD%E5%AE%8C%E6%95%B4%E7%9A%84useEffect%E6%8C%87%E5%8D%97.md

File: src/pages/ActivityTicketDetailsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+      setSessions(activityTicketData.sessions)
+
+      setInvoice({
+        name: activityTicketData.invoice.name,
+        email: activityTicketData.invoice.email,
+        phone: activityTicketData.invoice.phone,
+        orderProductId: activityTicketData.invoice.orderProductId,
+      })
+    }
+  }, [activityTicketData])
```
[the remark this answers is not in the document]
Comment (kkshyu) [reply src/pages/ActivityTicketDetailsPage.tsx]: can also use `useReducer`

**Pull request merged**